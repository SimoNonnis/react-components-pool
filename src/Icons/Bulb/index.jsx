import React from 'react';
import PropTypes from 'prop-types';

/**
 * Bulb Icon.
 * @param {number} width
 * @param {number} height
 * @param {string} fill
 * @param {string} className
 * @param {string} classNamePath
 * @returns {React.Component}
 */
function Bulb({ width, height, fill, className, classNamePath }) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 100 100"
      width={width}
      height={height}
      className={className}
    >
      <title>Bulb Icon</title>

      <path
        className={classNamePath}
        fill={fill}
        d="M50,7.43A29.38,29.38,0,0,0,31,59.21a14.5,14.5,0,0,1,3.49,8.3,2.35,2.35,0,0,0,2.33,2.07H63.18a2.34,2.34,0,0,0,2.33-2.07c.29-2.41,1.1-5.79,3.35-8.17A29.38,29.38,0,0,0,50,7.43ZM74.69,36.81a24.82,24.82,0,0,1-9,19.06l-.08.07a18.3,18.3,0,0,0-4.44,8.94H38.81a18.38,18.38,0,0,0-4.28-8.77,2.33,2.33,0,0,0-.34-.35,24.69,24.69,0,1,1,40.51-19Z"
      />
      <path
        className={classNamePath}
        fill={fill}
        d="M63.18,72.55H36.81a2.35,2.35,0,0,0,0,4.7H63.18a2.35,2.35,0,0,0,0-4.7Z"
      />
      <path
        className={classNamePath}
        fill={fill}
        d="M63.18,80.21H36.81a2.35,2.35,0,0,0,0,4.7H63.18a2.35,2.35,0,0,0,0-4.7Z"
      />
      <path
        className={classNamePath}
        fill={fill}
        d="M55.52,87.88h-11a2.35,2.35,0,1,0,0,4.7h11a2.35,2.35,0,1,0,0-4.7Z"
      />
    </svg>
  );
}

Bulb.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
  fill: PropTypes.string,
  className: PropTypes.string,
  classNamePath: PropTypes.string,
};

Bulb.defaultProps = {
  width: 40,
  height: 40,
  fill: 'currentColor',
  className: 'bulbIcon',
  classNamePath: 'bulbIconPath',
};

export default Bulb;
