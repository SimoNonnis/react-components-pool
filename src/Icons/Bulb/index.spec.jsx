/* eslint-env jest */
import React from 'react';

import Bulb from './';

describe('Test <Bulb /> icon', () => {
  it('should render Bulb icon correctly with default props', () => {
    const wrapper = shallow(<Bulb />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Bulb icon correctly with props', () => {
    const wrapper = shallow(
      <Bulb fill="#a4a1a1" className="testClassname" height={30} width={30} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
