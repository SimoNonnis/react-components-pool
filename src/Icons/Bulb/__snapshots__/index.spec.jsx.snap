// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Bulb /> icon should render Bulb icon correctly with default props 1`] = `
<svg
  className="bulbIcon"
  height={40}
  viewBox="0 0 100 100"
  width={40}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Bulb Icon
  </title>
  <path
    className="bulbIconPath"
    d="M50,7.43A29.38,29.38,0,0,0,31,59.21a14.5,14.5,0,0,1,3.49,8.3,2.35,2.35,0,0,0,2.33,2.07H63.18a2.34,2.34,0,0,0,2.33-2.07c.29-2.41,1.1-5.79,3.35-8.17A29.38,29.38,0,0,0,50,7.43ZM74.69,36.81a24.82,24.82,0,0,1-9,19.06l-.08.07a18.3,18.3,0,0,0-4.44,8.94H38.81a18.38,18.38,0,0,0-4.28-8.77,2.33,2.33,0,0,0-.34-.35,24.69,24.69,0,1,1,40.51-19Z"
    fill="currentColor"
  />
  <path
    className="bulbIconPath"
    d="M63.18,72.55H36.81a2.35,2.35,0,0,0,0,4.7H63.18a2.35,2.35,0,0,0,0-4.7Z"
    fill="currentColor"
  />
  <path
    className="bulbIconPath"
    d="M63.18,80.21H36.81a2.35,2.35,0,0,0,0,4.7H63.18a2.35,2.35,0,0,0,0-4.7Z"
    fill="currentColor"
  />
  <path
    className="bulbIconPath"
    d="M55.52,87.88h-11a2.35,2.35,0,1,0,0,4.7h11a2.35,2.35,0,1,0,0-4.7Z"
    fill="currentColor"
  />
</svg>
`;

exports[`Test <Bulb /> icon should render Bulb icon correctly with props 1`] = `
<svg
  className="testClassname"
  height={30}
  viewBox="0 0 100 100"
  width={30}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Bulb Icon
  </title>
  <path
    className="bulbIconPath"
    d="M50,7.43A29.38,29.38,0,0,0,31,59.21a14.5,14.5,0,0,1,3.49,8.3,2.35,2.35,0,0,0,2.33,2.07H63.18a2.34,2.34,0,0,0,2.33-2.07c.29-2.41,1.1-5.79,3.35-8.17A29.38,29.38,0,0,0,50,7.43ZM74.69,36.81a24.82,24.82,0,0,1-9,19.06l-.08.07a18.3,18.3,0,0,0-4.44,8.94H38.81a18.38,18.38,0,0,0-4.28-8.77,2.33,2.33,0,0,0-.34-.35,24.69,24.69,0,1,1,40.51-19Z"
    fill="#a4a1a1"
  />
  <path
    className="bulbIconPath"
    d="M63.18,72.55H36.81a2.35,2.35,0,0,0,0,4.7H63.18a2.35,2.35,0,0,0,0-4.7Z"
    fill="#a4a1a1"
  />
  <path
    className="bulbIconPath"
    d="M63.18,80.21H36.81a2.35,2.35,0,0,0,0,4.7H63.18a2.35,2.35,0,0,0,0-4.7Z"
    fill="#a4a1a1"
  />
  <path
    className="bulbIconPath"
    d="M55.52,87.88h-11a2.35,2.35,0,1,0,0,4.7h11a2.35,2.35,0,1,0,0-4.7Z"
    fill="#a4a1a1"
  />
</svg>
`;
